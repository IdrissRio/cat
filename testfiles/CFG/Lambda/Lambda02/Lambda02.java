
public class Lambda02 {
  void foo() {
    Integer x = 0;
    Runnable r = () -> { System.out.println(x); };
    r.run();
  }
}