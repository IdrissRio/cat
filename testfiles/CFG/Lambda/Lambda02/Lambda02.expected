NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [IntegerLiteral[46]]
NodeName: IntegerLiteral[46] Successors: [VariableDeclarator[43]]
NodeName: VariableDeclarator[43] Successors: [LambdaDefinition[56]]
NodeName: LambdaDefinition[56] Successors: [VariableDeclarator[52]]
NodeName: VariableDeclarator[52] Successors: [VarAccess[76]]
NodeName: VarAccess[76] Successors: [MethodAccess[77]]
NodeName: MethodAccess[77] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []
NodeName: Entry[57] Successors: [TypeAccess[66]]
NodeName: TypeAccess[66] Successors: [VarAccess[67]]
NodeName: VarAccess[67] Successors: [VarAccess[71]]
NodeName: VarAccess[71] Successors: [MethodAccess[68]]
NodeName: MethodAccess[68] Successors: [Exit[74]]
NodeName: Exit[74] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [MethodAccess[77]]
NodeName: MethodAccess[77] Predecessor: [VarAccess[76]]
NodeName: VarAccess[76] Predecessor: [VariableDeclarator[52]]
NodeName: VariableDeclarator[52] Predecessor: [LambdaDefinition[56]]
NodeName: LambdaDefinition[56] Predecessor: [VariableDeclarator[43]]
NodeName: VariableDeclarator[43] Predecessor: [IntegerLiteral[46]]
NodeName: IntegerLiteral[46] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: Exit[74] Predecessor: [MethodAccess[68]]
NodeName: MethodAccess[68] Predecessor: [VarAccess[71]]
NodeName: VarAccess[71] Predecessor: [VarAccess[67]]
NodeName: VarAccess[67] Predecessor: [TypeAccess[66]]
NodeName: TypeAccess[66] Predecessor: [Entry[57]]
NodeName: Entry[57] Predecessor: []
